### PR TITLE
Cleanup filtering (in prep for search on breakdown page)

### PR DESCRIPTION
**Pull request by Arkham**

### CODE REVIEW

File: nri-log-explorer/src/Main.hs [hunk 1]
```diff
@@ -697,60 +713,65 @@ handleEvent ::
...
-              (ListWidget.handleListEventVi ListWidget.handleListEvent vtyEvent)
-              model
-              |> andThen Brick.continue
+    (Brick.VtyEvent vtyEvent) -> do
+      case (toMode model, vtyEvent) of
+        -- Quitting
+        (NormalMode, Vty.EvKey (Vty.KChar 'q') []) -> do Brick.halt model
+        (_, Vty.EvKey (Vty.KChar 'c') [Vty.MCtrl]) -> Brick.halt model
+        -- Navigation
+        (_, Vty.EvKey Vty.KEnter []) -> do
```
Comment (stoeffel) [inline nri-log-explorer/src/Main.hs:722]: `_` ==> has meaning in either mode.

File: nri-log-explorer/src/Main.hs [hunk 2]
```diff
@@ -697,60 +713,65 @@ handleEvent ::
...
-          -- Fallback
-          _ ->
-            scroll
-              (ListWidget.handleListEventVi ListWidget.handleListEvent vtyEvent)
-              model
-              |> andThen Brick.continue
+    (Brick.VtyEvent vtyEvent) -> do
+      case (toMode model, vtyEvent) of
+        -- Quitting
+        (NormalMode, Vty.EvKey (Vty.KChar 'q') []) -> do Brick.halt model
```
Comment (stoeffel) [inline nri-log-explorer/src/Main.hs:719]: `NormalMode` => we don't want this to halt when we are editing.

File: nri-log-explorer/src/Main.hs [hunk 3]
```diff
@@ -218,16 +248,24 @@ update model msg =
...
-                    EditFilter previous _ -> EditFilter previous editor
-                    HasFilter first rest ->
-                      EditFilter (Just (first, rest))
-                        <| Edit.applyEdit (\_ -> TZ.gotoEOL <| TZ.textZipper [Text.join " " (first : rest)] (Just 1)) editor
-              }
+            ( \page ->
+                case page of
+                  NoDataPage _ -> Prelude.pure page
+                  SpanBreakdownPage _ _ _ -> Prelude.pure page
+                  RootSpanPage time filter spans ->
```
Comment (stoeffel) [inline nri-log-explorer/src/Main.hs:257]: `/` Shouldn't display the edit field (just yet) on any other page.

File: nri-log-explorer/src/Main.hs [hunk 4]
```diff
@@ -92,14 +92,12 @@ data Span = Span
 data Msg
   = AddRootSpan ByteString.ByteString
   | ShowDetails
-  | Exit
+  | Confirm
+  | Cancel
```
Comment (jwoudenberg-nri) [inline nri-log-explorer/src/Main.hs:96]: 😻

File: nri-log-explorer/src/Main.hs [hunk 5]
```diff
@@ -198,11 +196,43 @@ update model msg =
...
-        }
-        |> continueAfterUserInteraction
+    Confirm ->
+      withPage
+        model
+        ( \page ->
+            case page of
+              NoDataPage _ -> Prelude.pure page
+              SpanBreakdownPage _ _ _ -> Prelude.pure page
+              RootSpanPage time (EditFilter _ filterEditor) spans ->
```
Comment (jwoudenberg-nri) [inline nri-log-explorer/src/Main.hs:206]: Elegant solution!

File: nri-log-explorer/src/Main.hs [hunk 6]
```diff
@@ -697,60 +713,65 @@ handleEvent ::
...
-            liftIO (pushMsg ClearFilter)
-            Brick.continue model
-          -- Fallback
-          _ ->
-            scroll
-              (ListWidget.handleListEventVi ListWidget.handleListEvent vtyEvent)
-              model
-              |> andThen Brick.continue
+    (Brick.VtyEvent vtyEvent) -> do
+      case (toMode model, vtyEvent) of
```
Comment (jwoudenberg-nri) [inline nri-log-explorer/src/Main.hs:717]: 😻 

Also: missing a command mode that doesn't do much, but requires you typing `:visual` to get out of it again.

**Review by jwoudenberg-nri — APPROVED** [review]
Love it, thank you!

**Pull request merged**